src/components/lib/jitutils/jitutils.h

```diff
@@ -0,0 +1,77 @@
+#include "cos_types.h"
+#include "cos_debug.h"
+
+static int
+jitutils_search(u8_t *src, u8_t *pat, size_t len, size_t max)
+{
+	unsigned int i, j;
+
+	/* naive pattern search */
+	for (i = 0; i < max; i++) {
+		for (j = 0; j < len; j++) {
+			if (src[i + j] != pat[j]) break;
+		}
+ 
+		if (j == len) return i;
+	}
+
+	return -1;
+}
+
+static int 
+jitutils_replace(u8_t *src, u8_t *orig, u8_t *replace, size_t len, size_t src_len)
+{
+	unsigned int i;
+	int pos;
+
+	pos = jitutils_search(src, orig, len, src_len);
+	if (pos == -1) {
+		for (i = 0; i < src_len; i++) { 
+			printc("%x ", src[i]);
+		}
+
+		printc("\n");
+		return 0;
+	}
+
+	for (i = 0; i < len; i++) {
+		src[pos + i] = replace[i];
+	}
+	
+	return pos;
+}
+
+#define JIT_TOK_PLACEHOLDER   0xdeadbeefdeadbeeful
```

Comment:
use `UL` rather than `ul` seems to be the convention of the code base.

Reply:
Agreed.